vpr/src/base/atom_netlist.cpp

```diff
@@ -141,8 +141,9 @@ AtomBlockId AtomNetlist::create_block(const std::string& name, const t_model* mo
     AtomBlockId blk_id = Netlist::create_block(name);
 
     //Initialize the data
-    block_models_.push_back(model);
-    block_truth_tables_.push_back(truth_table);
+    // Use insert instead of push_back
```

Comment:
Can you please make this comment a bit more clear. Something like "push back new data or overwrite data for block". This makes it clear that the next lines are both initializing the data and overwriting it in this line. There is also no need to mention that we are using insert instead of push_back since, after your change, the context of this comment will no longer make sense.

For this, and the other functions you fixed, could you please update the Doxygen comment to reflect this change. It should mention that when an existing block is found, the data is overwritten. Its a good idea to document this stuff well.
https://github.com/verilog-to-routing/vtr-verilog-to-routing/blob/33a82f1c2f0b1a8e95c0efb06691b0d54ecfa629/vpr/src/base/atom_netlist.h#L167-L176